### PR TITLE
Typing: Narrow down types of arguments (NDFrame) #10

### PR DESCRIPTION
**Pull request by mKlepsch**
[Typing: Narrow down types of arguments (NDFrame) #10](https://github.com/noatamir/pydata-workshop/issues/10)
Made these specifications to Literals:
- [ X] Specify date_unit in to_json more precisely
- [ X] Specify mode in to_hdf more precisely
- [ X] Specify complib in to_hdf more precisely
- [ X] Specify format in to_hdf more precisely

@phofl @jorisvandenbossche @noatamir @MarcoGorelli

### CODE REVIEW

File: pandas/core/generic.py [hunk 1]
```diff
@@ -2276,7 +2276,7 @@ def to_json(
         date_format: str | None = None,
         double_precision: int = 10,
         force_ascii: bool_t = True,
-        date_unit: str = "ms",
+        date_unit: Literal["s", "ms", "us", "ns"] = "ms",
```
Comment (MarcoGorelli) [inline pandas/core/generic.py:2279]: this one seems broadly useful - perhaps let's define it as `TimeUnit` in `_typing.py` and import it here?

File: pandas/core/generic.py [hunk 2]
```diff
@@ -2556,11 +2556,11 @@ def to_hdf(
         self,
         path_or_buf: FilePath | HDFStore,
         key: str,
-        mode: str = "a",
+        mode: Literal["a", "w", "r+"] = "a",
         complevel: int | None = None,
-        complib: str | None = None,
+        complib: ["zlib", "lzo", "bzip2", "blosc"] | None = None,
```
Comment (MarcoGorelli) [inline pandas/core/generic.py:2561]: missing literal

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
thanks for your pr

**Comment by mKlepsch** [comment]
Hi, I can't figure out why it is still failing the CI. Could you give me some pointers?

**Comment by MarcoGorelli** [comment]
you'll need to run the pre-commit hooks, please check the contributing guide for how to do that

**Comment by phofl** [comment]
You can do:

```
pre-commit install
pre-commit run --all-files
```

(It's probably a space at the end of the line, which is disallowed

**Review by phofl — COMMENTED** [review]
Looks good after fixing pre-commit

**Review by mroeschke — APPROVED** [review]

**Review by phofl — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @mKlepsch

**Pull request merged**